### PR TITLE
299229: Fixes for local development

### PR DESCRIPTION
**Pull request by jeevankuduvaravindran**
# **What this PR does / why we need it**:
This PR contains the fixes for database migration for local development.

Added docker-compose.migrate.yaml

- Fixed migration scripts
- Fixed db-migration.dockerfile
- Updated pre-commit version

[AB#299229](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/299229)

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [ ] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: templates/adp-backend-template-node/skeleton/docker-compose.migrate.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    entrypoint: >
+      sh -c "/scripts/migration/database-down"
+    depends_on:
+      - ${{ values.project_name }}-postgres
+    volumes:
+      - ./changelog:/liquibase/changelog
+      - ./scripts:/scripts
+
+  ${{ values.project_name }}-postgres:
+    image: postgres:11.4-alpine
```
Comment (DaRoszko) [inline templates/adp-backend-template-node/skeleton/docker-compose.migrate.yaml:42]: shouldn't be 11 ?

Comment (DaRoszko) [reply templates/adp-backend-template-node/skeleton/docker-compose.migrate.yaml:42]: 15 / 16?

Comment (jeevankuduvaravindran) [reply templates/adp-backend-template-node/skeleton/docker-compose.migrate.yaml:42]: @DaRoszko It is the same as in our demo apps. Also, checked some of the ffc-pay and ffc-ahwr repos. They are using the same.

Comment (kbabigumira) [reply templates/adp-backend-template-node/skeleton/docker-compose.migrate.yaml:42]: @DaRoszko  one to discuss with John. Most services are still using 11.

File: templates/adp-backend-template-node/skeleton/.secrets.baseline [hunk 2]
```diff
@@ -104,6 +107,16 @@
...
     }
   ],
-  "results": {},
-  "generated_at": "2022-02-22T08:52:45Z"
+  "results": {
+    "docker-compose.yaml": [
+      {
+        "type": "Secret Keyword",
+        "filename": "docker-compose.yaml",
+        "hashed_secret": "b3054ff0797ff0b2bbce03ec897fe63e0b6490e0",
```
Comment (kbabigumira) [inline templates/adp-backend-template-node/skeleton/.secrets.baseline:115]: should this be checked in?

Comment (jeevankuduvaravindran) [reply templates/adp-backend-template-node/skeleton/.secrets.baseline:115]: @kbabigumira it is just on the docker-compose file for local development. There isn't anything sensitive on it. As per recommendation, we can set "is_verified": true to the detected secret.

**Review by DaRoszko — APPROVED** [review]

**Review by kbabigumira — APPROVED** [review]

**Pull request merged**